src/file.cpp

```diff
@@ -509,7 +509,14 @@ void SyncFileGet::updatelocalname()
 // add corresponding LocalNode (by path), then self-destruct
 void SyncFileGet::completed(Transfer*, LocalNode*)
 {
-    sync->checkpath(NULL, &localname);
+    LocalNode *ll = sync->checkpath(NULL, &localname);
+    if (ll && ll != (LocalNode*)~0)
+    {
+        LOG_debug << "LocalNode created, associating with remote Node";
+        ll->setnode(n);
```

Comment:
It would be good to check that the fingerprint, timestamp... of the localnode matches the ones in the node. Otherwise, it would mean that the file has been modified externally in the meantime.